File: backend/app/models/counterparty.py
from datetime import datetime

from sqlalchemy import DateTime, ForeignKey, Integer, String, func
from sqlalchemy.orm import Mapped, mapped_column

from app.db import Base


class Counterparty(Base):
    __tablename__ = "counterparties"

    id: Mapped[int] = mapped_column(Integer, primary_key=True)
    company_id: Mapped[int] = mapped_column(
        Integer, ForeignKey("companies.id"), nullable=False
    )
    name: Mapped[str] = mapped_column(String, nullable=False)
    eik: Mapped[str] = mapped_column(String, nullable=False)
    vat_number: Mapped[str | None] = mapped_column(String, nullable=True)
    address: Mapped[str] = mapped_column(String, nullable=False)
    mol: Mapped[str | None] = mapped_column(String, nullable=True)
    email: Mapped[str | None] = mapped_column(String, nullable=True)
    phone: Mapped[str | None] = mapped_column(String, nullable=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime, nullable=False, server_default=func.now()
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime, nullable=False, server_default=func.now(), onupdate=func.now()
    )

File: frontend/src/types/counterparty.ts
export interface Counterparty {
  id: number
  name: string
  eik: string
  vat_number: string | null
  address: string
  mol: string | null
  email: string | null
  phone: string | null
  created_at: string
  updated_at: string
}

export interface CounterpartyInput {
  name: string
  eik: string
  vat_number: string | null
  address: string
  mol: string | null
  email: string | null
  phone: string | null
}
